### DISCUSSION TITLE
getSimplePath get path of  when back ward state is pointed , last transition event is missing

### DISCUSSION THREAD

**Discussion by prashnatgthorat** [statelyai/xstate#2763] [Q&A]
Hello Team, 
i am facing issue to generate path for when the transition is pointing to backward state , then that transition event is not capture in path.

Able to generate Generated path 
**Add Log -> save -> submitted -> done**
**Add Log -> save -> submitted -> done -> delete**

Expected Paths to be generate
**Add Log -> save -> submitted -> done -> edit**
**Add Log -> save -> submitted -> done -> delete -> Cancel**

Please help me how i can get such paths , what is config i need to pass in getSimplePath OR is this known issue?

Below is the State machine JSON 
![transitionPointtoBackward](https://user-images.githubusercontent.com/93247803/139022389-fd0bea94-d354-4dd3-946d-790780cee360.png)

stateMachineJson

{
  "id": "StateMachine",
  "initial": "home",
  "context": {
    "retries": 0
  },
  "states": {
    "home": {
      "on": {
        "ADD LOG": "Blood Sugar"
      }
    },
   
    "Blood Sugar": {
      "on": {
        "SAVE": "submitted",
      }
    },
     "submitted": {
      "on": {
        "done": "Test Blood sugar",
       
      }
    },
    "Test Blood sugar": {
      "on": {
        "edit": "Blood Sugar",
        "delete": "Delete popup"
      }
    },
    "Delete popup": {
      "on": {
        "CANCEL": "Test Blood sugar"
      }
    }
  }
}

**Comment by davidkpiano** [comment]
The algorithm works by traversing paths that don't repeat states, so to do this, you can add a child state:

```js
'Test Blood sugar': {
  // ...
  initial: 'default',
  states: {
    default: {},
    'after cancel': {}
  }
},
'Delete popup': {
  on: { CANCEL: 'Test Blood sugar.afterCancel' }
}
```
